deepchecks/vision/checks/data_integrity/abstract_property_outliers.py

```diff
@@ -57,51 +58,44 @@ class AbstractPropertyOutliers(SingleDatasetCheck):
         Two percentiles which define the IQR range
     iqr_scale: float, default: 1.5
         The scale to multiply the IQR range for the outliers detection
+    properties_precalculated: bool, default: False
+        Are the properties passed to the check precalculated
     """
 
     def __init__(self,
-                 properties: t.List[t.Dict[str, t.Any]] = None,
+                 properties_list: t.List[t.Dict[str, t.Any]] = None,
+                 property_input_type: PropertiesInputType = PropertiesInputType.OTHER,
                  n_show_top: int = 5,
                  iqr_percentiles: t.Tuple[int, int] = (25, 75),
                  iqr_scale: float = 1.5,
                  **kwargs):
         super().__init__(**kwargs)
-        if properties is not None:
-            self.user_properties = label_prediction_properties.validate_properties(properties)
-            # Validate no property have class_id as output_type
-            if any(p['output_type'] == 'class_id' for p in self.user_properties):
-                raise DeepchecksValueError('Properties cannot have class_id as output_type for outliers checks')
-        else:
-            self.user_properties = None
-
+        self.properties_list = properties_list
+        self.property_input_type = property_input_type
         self.iqr_percentiles = iqr_percentiles
         self.iqr_scale = iqr_scale
         self.n_show_top = n_show_top
 
         self._properties_results = None
-        self._properties_funcs = None
 
     def initialize_run(self, context: Context, dataset_kind: DatasetKind):
         """Initialize the properties state."""
-        self._properties_results = defaultdict(list)
         data = context.get_data_by_kind(dataset_kind)
 
+        self._properties_results = defaultdict(list)
         # Take either alternative properties if defined or default properties defined by the child class
-        if self.user_properties is not None:
-            self._properties_funcs = self.user_properties
-        else:
-            self._properties_funcs = self.get_default_properties(data)
-            # Filter out properties that have class_id as output_type
-            self._properties_funcs = [p for p in self._properties_funcs if p['output_type'] != 'class_id']
+        self.properties_list = self.properties_list if self.properties_list else self.get_default_properties(data)
+        if any(p['output_type'] == 'class_id' for p in self.properties_list):
+            warnings.warn('Properties that have class_id as output_type will be skipped.')
```

Comment:
Why? Because they don't make sense in the context of outliers?

Reply:
Yes, the outliers on the class id are meaningless because it's only encoding of non-ordinal data